plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -214,47 +214,101 @@ private void returnFailure(BulkByScrollResponse response, String modelId, Action
         actionListener.onFailure(new OpenSearchStatusException(errorMessage, RestStatus.INTERNAL_SERVER_ERROR));
     }
 
-    private void deleteModel(String modelId, ActionListener<DeleteResponse> actionListener) {
-        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
-        client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
+    private void deleteModel(String modelId, FunctionName functionName, ActionListener<DeleteResponse> actionListener) {
+        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId).setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        client.delete(deleteRequest, new ActionListener<>() {
             @Override
             public void onResponse(DeleteResponse deleteResponse) {
-                deleteModelChunks(modelId, deleteResponse, actionListener);
-                deleteController(modelId);
+                deleteModelChunksAndController(functionName, actionListener, modelId, deleteResponse);
             }
 
             @Override
             public void onFailure(Exception e) {
-                log.error("Failed to delete model meta data for model: " + modelId, e);
                 if (e instanceof ResourceNotFoundException) {
-                    deleteModelChunks(modelId, null, actionListener);
-                    deleteController(modelId);
+                    deleteModelChunksAndController(functionName, actionListener, modelId, null);
+                } else {
+                    log.error("Failed to delete model meta data for model, please try again: " + modelId, e);
+                    actionListener.onFailure(e);
                 }
-                actionListener.onFailure(e);
             }
         });
     }
 
+    private void deleteModelChunksAndController(
+        FunctionName functionName,
+        ActionListener<DeleteResponse> actionListener,
+        String modelId,
+        DeleteResponse deleteResponse
+    ) {
+        if (FunctionName.REMOTE != functionName) {
```

Comment:
How about simplify this just always delete chunks and controllers?

Reply:
We know remote model doesn't have chunks, the query to it's its chunk can be saved to improve a little bit performance. But I am ok to remove this check if you prefer to not check this.

Reply:
I think adding another delete won't bring a lot impact for performance, it can make the code easier to maintain without such if else